mqtt/mqtt-broker/tests/translation.rs

```diff
@@ -0,0 +1,264 @@
+        .build();
+
+    // device subscribes to twin update
+    device_1
+        .subscribe("$iothub/twin/PATCH/properties/desired/#", QoS::AtLeastOnce)
+        .await;
+
+    // Core sends update
+    edge_hub_core
+        .publish_qos1("$edgehub/device_1/twin/desired/?version=30", "", false)
+        .await;
+
+    // device recieves response
+    recieve_with_topic(
+        &mut device_1,
+        "$iothub/twin/PATCH/properties/desired/?version=30",
+    )
+    .await;
+
+    server_handle.shutdown().await;
+}
+
+// https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#respond-to-a-direct-method
+#[tokio::test]
+async fn translation_direct_method_response() {
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let mut server_handle = common::start_server(broker);
+
+    let mut edge_hub_core = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("edge_hub_core".into()))
+        .build();
+    let mut device_1 = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("device_1".into()))
+        .build();
+
+    // Core subscribes
+    edge_hub_core
+        .subscribe("$edgehub/+/methods/res/#", QoS::AtLeastOnce)
+        .await;
+
+    // device subscribes to direct methods
+    device_1
+        .subscribe("$iothub/methods/POST/#", QoS::AtLeastOnce)
+        .await;
+
+    // Core calls method
+    edge_hub_core
+        .publish_qos1(
+            "$edgehub/device_1/methods/post/my_cool_method/?rid=7",
+            "",
+            false,
+        )
+        .await;
+
+    // device recieves call and responds
+    recieve_with_topic(&mut device_1, "$iothub/methods/POST/my_cool_method/?rid=7").await;
+    device_1
+        .publish_qos1("$iothub/methods/res/200/?rid=7", "", false)
+        .await;
+
+    // Core recieves response
+    recieve_with_topic(
+        &mut edge_hub_core,
+        "$edgehub/device_1/methods/res/200/?rid=7",
+    )
+    .await;
+
+    server_handle.shutdown().await;
+}
+
+#[tokio::test]
+async fn translation_twin_notify() {
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let mut server_handle = common::start_server(broker);
+
+    let mut edge_hub_core = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("edge_hub_core".into()))
+        .build();
+    let mut device_1 = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("device_1".into()))
+        .build();
+
+    // Core subscribes
+    edge_hub_core
+        .subscribe("$edgehub/+/twin/get/#", QoS::AtLeastOnce)
+        .await;
+    edge_hub_core
+        .subscribe("$edgehub/subscriptions/device_1", QoS::AtLeastOnce)
```

Comment:
Nope. eh core will subscribe to $edgehub/connections to get all connected devices. Then for each connected device that has an iothub identity (except itself) it will subscribe to $edgehub/subscriptions/{clientid}